File: app/src/main/java/com/stevenfrew/beatprompter/PlaylistNode.java
package com.stevenfrew.beatprompter;

import com.stevenfrew.beatprompter.cache.SongFile;

public class PlaylistNode
{
    public SongFile mSongFile;
    PlaylistNode mNextNode;
    PlaylistNode mPrevNode;

    PlaylistNode(SongFile song)
    {
        mSongFile=song;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/set/PlaylistNode.kt
package com.stevenfrew.beatprompter.set

import com.stevenfrew.beatprompter.cache.SongFile
import com.stevenfrew.beatprompter.song.SongInfoProvider

data class PlaylistNode(
	val songFile: SongFile,
	val variation: String? = null,
	val nextSong: PlaylistNode? = null
) : SongInfoProvider {
	override val songInfo = songFile
}
